src/requirements.ts

```diff
@@ -66,33 +59,54 @@ function checkJavaRuntime(): Promise<string> {
         }
         invalidJavaHome(reject, msg);
       }
-      return resolve(javaHome);
+      resolve(javaHome);
     }
-    //No settings, let"s try to detect as last resort.
+    // No settings, check if we have a managed one
+    javaHome = context.globalState.get(JAVA_MANAGED_HOME_KEY, null);
+    if (javaHome) {
+      resolve(javaHome);
```

Comment:
Maybe (not sure) you need to add a return, or your program here will continue and you may call `resolve` or `reject` after the promise has been `resolve`d here.